### PR TITLE
Add next() builtin implementation and tests.

### PR DESCRIPTION
**Pull request by ericatkin**

### CODE REVIEW

File: project.properties [hunk 1]
```diff
@@ -12,4 +12,4 @@
 
 android.library=true
 # Project target.
-target=android-14
+target=android-23
```
Comment (freakboy3742) [inline project.properties]: Not completely sure of the android etiquette here - is there a reason to keep this "up to date"? I originally thought the best approach was "keep this number as low as possible" for backwards compatibility reasons.

File: python/common/org/Python.java [hunk 2]
```diff
@@ -1005,10 +1005,24 @@ public static void initializeModule(java.lang.Class cls, java.util.Map<java.lang
...
             "Return the next item from the iterator. If default is given and the iterator\n" +
-            "is exhausted, it is returned instead of raising StopIteration.\n"
+            "is exhausted, it is returned instead of raising StopIteration.\n",
+        args = {"iterator"},
+        default_args = {"_default"}
     )
-    public static org.python.Object next() {
-        throw new org.python.exceptions.NotImplementedError("Builtin function 'next' not implemented");
+    public static org.python.Object next(org.python.Object iterator, org.python.Object _default) {
+        if (iterator instanceof org.python.Iterable) {
```
Comment (freakboy3742) [inline python/common/org/Python.java]: `instanceof` works; but if we can wrap this into the `utils.isinstance` mechanism, that would be even better.

File: project.properties (hunk 1 above)
Comment (ericatkin) [inline project.properties]: 23 is all I had. I wasn't sure either. I have no problem reverting to 14.

File: python/common/org/Python.java (hunk 2 above)
Comment (freakboy3742) [inline python/common/org/Python.java]: Scratch that - forgot this is VOC, not Batavia.

**Pull request merged**